vita3k/regmgr/src/regmgr.cpp

```diff
@@ -58,15 +58,41 @@ static std::string decryptRegistryFile(const fs::path reg_path) {
     return res;
 }
 
+enum RegType {
+    REG_TYPE_INT,
+    REG_TYPE_STR,
+    REG_TYPE_BIN,
+};
+
 struct RegValue {
     std::string name;
+    RegType type;
     uint32_t size;
-    std::string value;
+    std::vector<char> init_value;
 };
 
 static std::vector<std::string> reg_category_template;
 static std::map<std::string, std::vector<RegValue>> reg_template;
 
+static std::vector<char> convert_int_to_hex_string(const int32_t &value) {
+    // Convert the int to a hex value
+    std::stringstream ss;
+    ss << std::hex << value;
+    std::string hex(ss.str());
+
+    // Add the zeros to the hex value when needed (4 bytes)
+    hex = std::string(8 - hex.length(), '0') + hex;
+
+    // Convert the hex value to a hex string (bytes)
+    std::string res;
+    for (size_t i = 0; i < hex.size(); i += 2) {
+        const int byte_value = std::stoi(hex.substr(i, 2), nullptr, 16);
+        res += static_cast<char>(byte_value);
+    }
+
+    return std::vector<char>(res.begin(), res.end());
+}
+
```

Comment:
wtf is that?

Reply:
where is problem ???

Reply:
```suggestion
static std::vector<char> convert_int_to_hex_string(const int32_t &value) {
    std::vector<char> res(4);
    *reinterpret_cast<int32_t*>(res.data()) = byte_swap(value);
    return res;
}

```

Reply:
Is see, i no know project have this func for that
thx